hs_restclient/__init__.py

```diff
@@ -1 +1,666 @@
+          u'resource_id': u'hr3hy35y5ht4y54hhthrtg43w',
+          u'resource_title': u'Other raster',
+          u'resource_type': u'RasterResource',
+          u'science_metadata_url': u'http://www.hydroshare.org/hsapi/scimeta/hr3hy35y5ht4y54hhthrtg43w/',
+          u'public': True}
+
+
+          Filtering (have):
+
+          /hsapi/resourceList/?from_date=2015-05-03&to_date=2015-05-06
+          /hsapi/resourceList/?user=admin
+          /hsapi/resourceList/?owner=admin
+          /hsapi/resourceList/?creator=admin
+          /hsapi/resourceList/?group=groupname
+          /hsapi/resourceList/?types=GenericResource&types=RasterResource
+
+          Filtering (need):
+
+          /hsapi/resourceList/?sharedWith=user
+
+        :raises:: HydroShareArgumentException if any filter parameters are invalid.
+        """
+        url = "{url_base}/resourceList/".format(url_base=self.url_base)
+
+        params = {}
+        if creator:
+            params['creator'] = creator
+        if owner:
+            params['owner'] = owner
+        if user:
+            params['user'] = user
+        if group:
+            params['group'] = group
+        if from_date:
+            if type(from_date) is datetime.date:
+                params['from_date'] = from_date.strftime('%Y-%m-%d')
+            else:
+                raise HydroShareArgumentException("from_date must of type '{0}'.".format(datetime.date))
+        if to_date:
+            if type(to_date) is datetime.date:
+                params['to_date'] = to_date.strftime('%Y-%m-%d')
+            else:
+                raise HydroShareArgumentException("to_date must of type '{0}'.".format(datetime.date))
+        if types:
+            if not is_sequence(types):
+                raise HydroShareArgumentException("Types must be a sequence type, but not a string.")
+            params['types'] = types
+
+        num_resources = 0
+
+        # Get first (only?) page of results
+        r = self._request('GET', url, params=params)
+        if r.status_code != 200:
+            raise HydroShareHTTPException((url, 'GET', r.status_code, params))
+        res = r.json()
+        tot_resources = res['count']
+        resources = res['results']
+        num_resources += len(res['results'])
+
+        for r in resources:
+            yield r
+
+        # Get remaining pages (if any exist)
+        while res['next'] and num_resources < tot_resources:
+            r = self._request('GET', res['next'], params=params)
+            if r.status_code != 200:
+                raise HydroShareHTTPException((url, 'GET', r.status_code, params))
+            res = r.json()
+            resources = res['results']
+            for r in resources:
+                yield r
+            num_resources += len(res['results'])
+
+        if num_resources != tot_resources:
+            raise HydroShareException("Expected {tot} resources but found {num}.".format(tot_resources, num_resources))
```

Comment:
This is kind of strange. You'll throw this exception from the top of a forloop *after* the entire loop has run. Are you sure you want that? It seems like strange behavior. I don't think its unusual for the number to be different. Maybe between the first request (where you find the total number of resources to expect) and the last page there have been new resources?

As a matter of fact, if there are more resources, or the pagination of them changes because of the order, do you do anything to make sure that works?